docs/CLN.md

```diff
@@ -0,0 +1,74 @@
+### Requirements 
+- CLN (complied with developer mode on) or LND
+- lspd
+- lspd plugin for cln
+- postgresql
+
+### Installation
+#### CLN 
+Follow compilation steps for CLN [here](https://github.com/ElementsProject/lightning/blob/master/doc/getting-started/getting-started/installation.md) to enable developer mode.
+
+#### lspd
+Needs to be build from source:
+```
+git clone https://github.com/breez/lspd 
+cd lspd
+go build .                                # compile lspd
+go build -o lspd_plugin ./cln_plugin/cmd  # compile lspd cln plugin
+```
+
+#### Postgresql
+Lspd supports postgresql backend. To create database and new role to access it on your postgres server use:
+##### Postgresql server
+```
+CREATE ROLE <username>;
+ALTER ROLE <username> WITH NOSUPERUSER INHERIT NOCREATEROLE NOCREATEDB LOGIN NOREPLICATION NOBYPASSRLS PASSWORD '<password>';
+CREATE DATABASE <dbname> WITH TEMPLATE = template0 ENCODING = 'UTF8' LC_COLLATE = 'en_US.UTF-8' LC_CTYPE = 'en_US.UTF-8';
+ALTER DATABASE <dbname> OWNER TO <username>;
+```
+##### RDS on AWS
+```
+CREATE ROLE <username>;
+ALTER ROLE <username> WITH INHERIT NOCREATEROLE NOCREATEDB LOGIN NOBYPASSRLS PASSWORD '<password>';
+CREATE DATABASE <dbname> WITH TEMPLATE = template0 ENCODING = 'UTF8' LC_COLLATE = 'en_US.UTF-8' LC_CTYPE = 'en_US.UTF-8';
+ALTER DATABASE <dbname> OWNER TO <username>;
+```
+
+### Configuration
+1. Create a random token (for instance using the command `openssl rand -base64 48`, or `./lspd genkey`)
+1. Define the environment variables as described in [sample.env](./sample.env). If `CERTMAGIC_DOMAIN` is defined, certificate for this domain is automatically obtained and renewed from Let's Encrypt. In this case, the port needs to be 443. If `CERTMAGIC_DOMAIN` is not defined, lspd needs to run behind a reverse proxy like treafik or nginx.
+
+ENV variables:
+- `LISTEN_ADDRESS` defines the host:port for the lspd grpc server
+- `CERTMAGIC_DOMAIN` domain on which lspd will be accessible
+- `DATABASE_URL` postgresql db url
+- `AWS_REGION` AWS region for SES emailing 
+- `AWS_ACCESS_KEY_ID` API key for SES emailing 
+- `AWS_SECRET_ACCESS_KEY`API secret  for SES emailing 
+- `MEMPOOL_API_BASE_URL` uses fee estimation for opening new channels (default: https://mempool.space)
+- `MEMPOOL_PRIORITY` priority with which open new channels using mempool api (default: economy)
+
+
+
+### Running lspd on CLN
+In order to run lspd on top of CLN, you need to run the lspd process and run cln with the provided cln plugin. You also need lightningd compiled with developer mode on (`./configure --enable-developer`)
+
+The cln plugin (go build -o lspd_plugin cln_plugin/cmd) is best started with a bash script to pass environment variables (note this LISTEN_ADDRESS is the listen address for communication between lspd and the plugin, this is not the listen address mentioned in the 'final step')
+
+```bash
+#!/bin/bash
+
+export LISTEN_ADDRESS=<listen address>
+/path/to/lspd_plugin
+```
```

Comment:
This startup of the plugin is out-of-date (I see it's out-of-date in the main readme too, sorry)

The cln plugin exposes a grpc interface that lspd uses to intercept htlcs.
The cln plugin is started by passing commandline parameters to cln, like so.
`lightningd --plugin=/path/to/lspd_plugin --lsp-listen=<grpc listen address>`

Optional:
- `--lsp-subscribertimeout=1m`: If lspd is not connected to the plugin, htlcs will 'hang'. If lspd hasn't connected within the subscriber timeout, the htlc is continued as if no lsp exists. Valid values are golang duration strings. (default: 1m)
- `--lsp-channel-accept-script`: Optional channel acceptor script for inbound channels. It's a starlark script that receives the `openchannel` or `openchannel2` request from cln, and needs to return a valid response for the hook. Example: `--lsp-channel-accept-script=(lambda oc: encode({"result":"continue"}) if oc["id"]=="031015a7839468a3c266d662d5bb21ea4cea24226936e2864a7ca4f2c3939836e0" else encode({"result":"reject", "error_message": "The channel cannot be opened"}))(decode(openchannel))`